src/braket/devices/device.py

```diff
@@ -33,7 +33,12 @@ def __init__(self, name: str, status: str):
 
     @abstractmethod
     def run(
-        self, task_specification: Union[Circuit, Problem], shots: Optional[int], *args, **kwargs
+        self,
+        task_specification: Union[Circuit, Problem],
+        shots: Optional[int] = 0,
```

Comment:
This I'm not as convinced by, since QPUs are also based on this. No strong opinions though.

Reply:
Then with that logic, task specification could be e.g. an annealing problem which won't take inputs, so maybe we shouldn't have defaults for either?